integration/spark/src/main/spark3/java/io/openlineage/spark3/agent/lifecycle/plan/Spark3EventHandlerFactory.java

```diff
@@ -0,0 +1,57 @@
+package io.openlineage.spark3.agent.lifecycle.plan;
+
+import com.google.common.collect.ImmutableList;
+import io.openlineage.client.OpenLineage;
+import io.openlineage.client.OpenLineage.DatasetFacet;
+import io.openlineage.client.OpenLineage.DatasetVersionDatasetFacet;
+import io.openlineage.client.OpenLineage.OutputDatasetFacet;
+import io.openlineage.spark.api.CustomFacetBuilder;
+import io.openlineage.spark.api.OpenLineageContext;
+import io.openlineage.spark.api.OpenLineageEventHandlerFactory;
+import io.openlineage.spark3.agent.facets.builder.DatasetVersionOutputDatasetFacetBuilder;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.BiConsumer;
+import lombok.AllArgsConstructor;
+import org.apache.spark.package$;
+
+public class Spark3EventHandlerFactory implements OpenLineageEventHandlerFactory {
+  @Override
+  public List<CustomFacetBuilder<?, ? extends OutputDatasetFacet>> createOutputDatasetFacetBuilders(
```

Comment:
Are there any tests for this class?

Reply:
[1) We have test that checks if service provider interface classes are properly loaded](https://github.com/OpenLineage/OpenLineage/blob/main/integration/spark/src/test/common/java/io/openlineage/spark/agent/lifecycle/InternalEventHandlerFactoryTest.java)
2) Integration tests check that we properly import and use this one by having proper output dataset facet there. 

Do you think we need unit test for that? I personally don't really know how that test would look like, as this class is useless without being included in OpenLineageRunEventBuilder.

DatasetVersionFacetConverter could be tested separately though...